how-to/workspace-platform-starter/client/src/modules/integrations/apps/integration.ts

```diff
@@ -621,44 +609,91 @@ export class AppProvider implements IntegrationModule<AppSettings> {
-						if (!lastApp && this._integrationHelpers.getApp && this._lastQuery === favInfo.command) {
-							lastApp = await this._integrationHelpers.getApp(favorite.typeId);
-						}
+			const { favoriteInfo } = await this.getFavInfo(FAVORITE_TYPE_NAME_APP);
 
-						if (!isEmpty(lastApp)) {
-							const rebuilt = await this.mapAppEntryToSearchEntry(
-								lastApp,
-								this._settings?.manifestTypeMapping,
-								favInfo,
-								payload.action === "set" ? favorite.id : undefined,
-								colorScheme
-							);
-
-							if (rebuilt) {
-								this._lastResponse.respond([rebuilt]);
-							}
-						}
+			const iconFolder = await this._integrationHelpers.getCurrentIconFolder();
+			const colorScheme = await this._integrationHelpers.getCurrentColorSchemeMode();
+
+			if (this._lastQuery === favoriteInfo?.command && payload.action === "delete") {
+				this._lastResponse.revoke(favorite.typeId);
+			} else if (this._lastAppResults) {
+				let lastApp = this._lastAppResults.find((a) => a.appId === favorite.typeId);
+
+				// If it wasn't in the last results add it, but only if we are in fav command
+				if (!lastApp && this._integrationHelpers?.getApp && this._lastQuery === favoriteInfo?.command) {
+					lastApp = await this._integrationHelpers.getApp(favorite.typeId);
+				}
+
+				if (!isEmpty(lastApp)) {
+					const rebuilt = await this.mapAppEntryToSearchEntry(
+						lastApp,
+						this._settings?.manifestTypeMapping,
+						iconFolder,
+						colorScheme,
+						favoriteInfo,
+						payload.action === "set" ? favorite.id : undefined
+					);
+
+					if (rebuilt) {
+						this._lastResponse.respond([rebuilt]);
 					}
 				}
 			}
 		}
 	}
+
+	/**
+	 * Get the favorite info and client if they are enabled.
+	 * @param favoriteTypeNames The type of client to get.
+	 * @returns The favorite info and client.
+	 */
+	private async getFavInfo(
+		favoriteTypeNames: FavoriteTypeNames
+	): Promise<{ favoriteClient: FavoriteClient | undefined; favoriteInfo: FavoriteInfo | undefined }> {
+		let favoriteInfo: FavoriteInfo | undefined;
+		let favoriteClient: FavoriteClient | undefined;
+
+		if ((this._definition?.data?.favoritesEnabled ?? true) && this._integrationHelpers?.getFavoriteClient) {
+			favoriteClient = await this._integrationHelpers.getFavoriteClient();
+			if (favoriteClient) {
+				favoriteInfo = favoriteClient.getInfo();
+				if (favoriteInfo.isEnabled) {
+					const isSupported =
+						isEmpty(favoriteInfo.enabledTypes) || favoriteInfo.enabledTypes.includes(favoriteTypeNames);
+					if (!isSupported) {
+						favoriteInfo = undefined;
+						favoriteClient = undefined;
+					}
+				}
+			}
+		}
+
+		return {
+			favoriteClient,
+			favoriteInfo
+		};
+	}
+
+	/**
+	 * Apply theming to an icon url.
+	 * @param url The url to theme.
+	 * @param iconFolder The icon folder.
+	 * @param colorSchemeMode The color scheme.
+	 * @returns The themed url.
+	 */
+	private themeUrl(
```

Comment:
Isn't this a function that was made importable? maybe we need to look at the option of introducing something like a getThemeClient which returns an object with all of these helper methods?

Reply:
I have separated all the theming methods to a ThemeClient, this is a breaking change, so MS365, SF and dev-extensions will also need to be updated.

Reply:
ok, but as discussed we are going to keep backwards compatibility for this release a log a warning to move to the theme client correct?